src/esphome/output/sm16716_output_component.cpp

```diff
@@ -0,0 +1,130 @@
+// Based on:
+//   - The SM16716 implementation from this repository.
+//   - The Tasmota SM16716 implementation:
+//     https://github.com/arendst/Sonoff-Tasmota/blob/master/sonoff/xdrv_04_light.ino
+//   - This older implementation:
+//     https://github.com/sowbug/sm16716
+//   - The datasheet (in Chinese only):
+//     https://github.com/sowbug/sm16716/blob/master/SM16716%20Datasheet%20%5BChinese%5D.pdf
+
+#include "esphome/defines.h"
+
+#ifdef USE_SM16716_OUTPUT
+
+#include "esphome/log.h"
+#include "esphome/output/sm16716_output_component.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace output {
+
+static const char *TAG = "output.sm16716";
+
+SM16716OutputComponent::SM16716OutputComponent(GPIOPin *pin_data, GPIOPin *pin_clock, uint8_t num_channels,
+                                               uint8_t num_chips, bool update)
+    : pin_data_(pin_data),
+      pin_clock_(pin_clock),
+      num_channels_(num_channels),
+      num_chips_(num_chips),
+      update_(update) {}
+
+void SM16716OutputComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up SM16716OutputComponent...");
+  this->pin_data_->setup();
+  this->pin_data_->digital_write(false);
+  this->pin_clock_->setup();
+  this->pin_clock_->digital_write(false);
+  this->pwm_amounts_.resize(this->num_channels_, 0);
+}
+void SM16716OutputComponent::dump_config() {
+  ESP_LOGCONFIG(TAG, "SM16716:");
+  LOG_PIN("  DATA Pin: ", this->pin_data_);
```

Comment:
```suggestion
  LOG_PIN("  Data Pin: ", this->pin_data_);
```

This shouts too much :)